ClassFormatError when trying to mock certain interfaces

Hi, 

I stumbled over some strange behaviour when trying to update from a 1.x version of Mockito to 2.x. One of my interfaces would no longer want to be mocked, throwing a MockitoException, with an error message of *Mockito cannot mock this class: interface &lt;MyInterfaceName&gt;*, ultimately triggered by a *java.lang.ClassFormatError: Duplicate field name&signature in class file &lt;MyInterfaceName&gt;$MockitoMock$1318839329*

This is the shortest code snippet I could come up with to reproduce the issue:

```java
import org.mockito.Mockito;

public class ClassFormatErrorTest {

    public static void main(String[] args) {
        // the next line will throw a MockitoException, caused by an
        // IllegalStateException, ultimately caused by ClassFormatError
        Mockito.mock(SimpleInterface.class);
    }

    private interface SimpleInterface {
        String getNameOfYs(X x, String str1, String str2, String str3);
        String getNameOfXs(Y y, String str1, String str2, String str3);
    }

    private class X {}
    private class Y {}
}
```

Software versions used:
* JVM version 1.8.0_151-b12 from Oracle (the issue also occurred with the version before)
* mockito-core 2.11.0 (also tried a few older versions, with the same result)
* on a Debian Linux system

Running above code creates the following output:

```
Exception in thread "main" org.mockito.exceptions.base.MockitoException: 
Mockito cannot mock this class: interface ClassFormatErrorTest$SimpleInterface.
Most likely it is due to mocking a private class that is not visible to Mockito

	at ClassFormatErrorTest.main(ClassFormatErrorTest.java:8)
Caused by: java.lang.IllegalStateException: Error invoking java.lang.ClassLoader#defineClass
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Direct.defineClass(ClassInjector.java:412)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.inject(ClassInjector.java:185)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:187)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:79)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:4456)
	at org.mockito.internal.creation.bytebuddy.SubclassBytecodeGenerator.mockClass(SubclassBytecodeGenerator.java:115)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:37)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:34)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:138)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:346)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:161)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:355)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:32)
	at org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker.createMockType(SubclassByteBuddyMockMaker.java:71)
	at org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker.createMock(SubclassByteBuddyMockMaker.java:42)
	at org.mockito.internal.creation.bytebuddy.ByteBuddyMockMaker.createMock(ByteBuddyMockMaker.java:25)
	at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:35)
	at org.mockito.internal.MockitoCore.mock(MockitoCore.java:65)
	at org.mockito.Mockito.mock(Mockito.java:1821)
	at org.mockito.Mockito.mock(Mockito.java:1734)
	... 1 more
Caused by: java.lang.ClassFormatError: Duplicate field name&signature in class file ClassFormatErrorTest$SimpleInterface$MockitoMock$1318839329
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Direct.defineClass(ClassInjector.java:408)
	... 20 more
```

Before I created this short, self contained, single class example, I tried with the interface and two classes as separate files, with the same result, so it is not a problem of inner classes.

:exclamation: note that the interface's method names are important, they have to end in the class names + "s", otherwise the problem does not show.
:exclamation: also note that the methods have to take 4 parameters (works perfectly with only 3)

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -4,7 +4,7 @@ ext {
 
 def versions = [:]
 
-versions.bytebuddy = '1.7.4'
+versions.bytebuddy = '1.7.7'
 
 libraries.junit4 = 'junit:junit:4.12'
 libraries.assertj = 'org.assertj:assertj-core:2.8.0'
@@ -17,4 +17,4 @@ libraries.bytebuddyandroid = "net.bytebuddy:byte-buddy-android:${versions.bytebu
 
 libraries.objenesis = 'org.objenesis:objenesis:2.6'
 
-libraries.asm = 'org.ow2.asm:asm:5.2'
+libraries.asm = 'org.ow2.asm:asm:6.0'

